modules/alloy/src/main/ts/ephox/alloy/ui/single/MenuSpec.ts

```diff
@@ -58,6 +60,23 @@ const make: CompositeSketchFactory<MenuDetail, MenuSpec> = (detail, components,
     AlloyEvents.run<MenuItemHoverEvent>(ItemEvents.hover(), (menu, simulatedEvent) => {
       const item = simulatedEvent.event.item;
       Highlighting.highlight(menu, item);
+    }),
+
+    // Enforce only a single radio menu item is toggled by finding any other toggled
+    // radio menu items and untoggling them when a certain item is toggled
+    AlloyEvents.run<MenuItemToggledEvent>(ItemEvents.toggled(), (menu, simulatedEvent) => {
+      const { item, state } = simulatedEvent.event;
+      if (state) {
+        // TODO: TINY-8812 - This ideally should be done in a way such that a menu can have multiple radio groups.
+        const checkedRadioItems = SelectorFilter.descendants(menu.element, '[role="menuitemradio"][aria-checked="true"]');
```

Comment:
Yeah, as you mention, it would be ideal to avoid this kind of DOM inspection. It probably looks like the kind of thing we should use Highlighting (with single select), but I think it already has another purpose (probably selection). I wonder if some sort of `Grouping` behaviour would be applicable here, but probably not. Anyway, I'm not giving you any better idea than you have already.

Reply:
The main improvement I would suggest is just moving it outside of the main spec definition to a separate function.

Reply:
Yeah, I suspect we need to look at a `Grouping` or similar sketcher rather than a behaviour here.

Reply:
Have split it out and added a comment... hopefully that's what you meant.